### PR TITLE
Prevent overwrite _sqlColumnAliases by sub-queries, in Phalcon\Mvc\Model\Query

### PR DESCRIPTION
**Pull request by alexbusu**
Hello!

* Type: bug fix
* Link to issue: 
  * https://github.com/phalcon/cphalcon/issues/13006
  * https://github.com/phalcon/cphalcon/issues/12548
  * https://github.com/phalcon/cphalcon/issues/1731

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [ ] I wrote some tests for this PR.

Small description of change:

In order to prevent the SQL Column Aliases override in _root_ select, by the sub-queries, I have scoped the sub-queries aliases into another entries in `_sqlColumnAliases`, which is now and array of set-of-aliases entries. The `_sqlColumnAliases[0]` contains the aliases of the root query, the `_sqlColumnAliases[1]` contains the set of aliases of the _last_ parsed sub-query on level 1, and so on.

Thanks

### CODE REVIEW

File: phalcon/mvc/model/query.zep [hunk 1]
```diff
@@ -178,6 +180,9 @@ class Query implements QueryInterface, InjectionAwareInterface
 		} else {
 			let this->_enableImplicitJoins = globals_get("orm.enable_implicit_joins");
 		}
+
+		let this->_nestingLevel = -1;
```
Comment (CameronHall) [inline phalcon/mvc/model/query.zep]: How about setting `-1` as the default value when you define the property on the class?

File: phalcon/mvc/model/query.zep [hunk 2]
```diff
@@ -178,6 +180,9 @@ class Query implements QueryInterface, InjectionAwareInterface
...
+		let this->_sqlColumnAliases = [];
```
Comment (CameronHall) [inline phalcon/mvc/model/query.zep]: Same here, set `[]` as the default property on the class.

**Review by CameronHall — COMMENTED** [review]
Hey @alexbusu,

Love your work! I've suggested two minor changes if you wanted to take a quick look :)

File: phalcon/mvc/model/query.zep (hunk 1 above)
Comment (alexbusu) [reply phalcon/mvc/model/query.zep]: Absolutely!

File: phalcon/mvc/model/query.zep (hunk 2 above)
Comment (alexbusu) [reply phalcon/mvc/model/query.zep]: Good point :)

**Review by CameronHall — APPROVED** [review]

**Review by CameronHall — CHANGES_REQUESTED** [review]
Can you please rebase and target 4.0.x and squash your commits? 3.4.2 is the final revision of 3.4.x unless there's a critical bug or security vulnerability.

And don't forget to update the changelog.

**Review by CameronHall — DISMISSED** [review]

**Review by niden — APPROVED** [review]

**Pull request merged**